src/cljs/org/broadinstitute/firecloud_ui/main.cljs

```diff
@@ -169,71 +161,36 @@
 (react/defc LoggedOut
   {:render
    (fn [{:keys [props]}]
-     [:div {:style {:display (when (:hidden? props) "none")}}
-      [:div {:style {:padding "50px 25px"}}
-       [:div {:style {:marginBottom "2em"}} (text-logo)]
-       [:div {:id "google-sign-in-button"}]
-       [:div {:style {:marginTop "1em"}} [RegisterLink]]
-       [:div {:style {:maxWidth 600 :paddingTop "2em" :fontSize "small"}}
-        [:p {:style {:fontWeight "bold"}} "WARNING NOTICE"]
-        [:p {}
-         "You are accessing a US Government web site which may contain information that must be "
-         "protected under the US Privacy Act or other sensitive information and is intended for "
-         "Government authorized use only."]
-        [:p {}
-         "Unauthorized attempts to upload information, change information, or use of this web site "
-         "may result in disciplinary action, civil, and/or criminal penalties. Unauthorized users "
-         "of this website should have no expectation of privacy regarding any communications or "
-         "data processed by this website."]
-        [:p {}
-         "Anyone accessing this website expressly consents to monitoring of their actions and all "
-         "communications or data transiting or stored on related to this website and is advised "
-         "that if such monitoring reveals possible evidence of criminal activity, NIH may provide "
-         "that evidence to law enforcement officials."]]]])})
-
-
-(def google-sign-in-scopes
-  ["profile"
-   "email"
-   "https://www.googleapis.com/auth/devstorage.full_control"
-   "https://www.googleapis.com/auth/compute"])
+     [:div {:style {:padding "50px 25px"}}
+      [:div {:style {:marginBottom "2em"}} (text-logo)]
+      [:div {}
+       [comps/Button {:text "Sign In"
+                      :href (str "/service/login?path="
+                                 (js/encodeURIComponent (let [hash (nav/get-hash-value)]
+                                                          (if (clojure.string/blank? hash)
+                                                            "workspaces"
```

Comment:
This is exactly how the UI works. If a blank path were allowed, and came back in the redirect, the UI would do the right thing.